### PR TITLE
Bundle stdlib procs at compile time and fix uplevel/upvar edge cases

### PR DESCRIPTION
**Pull request by bitwisecook**
## Summary

This PR adds compile-time bundling of referenced standard-library procedures and fixes several edge cases in `uplevel` and `upvar` handling.

### Stdlib Prelude Bundling

Adds `core/compiler/stdlib_prelude.py` which implements reference-driven bundling of Tcl standard-library procedures. When a program references a stdlib command (e.g., `parray`, `tcl_endOfWord`), the compiler now:

1. Reads the library's `tclIndex` to map commands to their source files
2. Lowers the referenced `.tcl` files to IR at compile time
3. Merges procedures into the module so they compile to WASM functions
4. Bundles only files for commands actually referenced (transitively)
5. Prunes unreachable bundled procs when safe (no dynamic dispatch)

This produces self-contained binaries that don't require `TCL_LIBRARY` at runtime. An allowlist (`DEFAULT_STDLIB_ALLOWLIST`) gates which commands seed bundling by default; transitive dependencies are always included. Fallback to runtime auto-loading occurs for:
- Commands outside the allowlist
- Non-lowerable library files
- Modules with dynamic dispatch (eval/source/dynamic names)

### Uplevel/Upvar Fixes

**Uplevel:**
- `uplevel #N` where N equals the current frame depth now correctly targets that frame's own locals (not the global frame)
- `uplevel -0` is now recognized as a valid relative level 0 (current frame)
- Negative levels like `uplevel -1` now properly raise `bad level` errors instead of treating them as body words
- Signed integer parsing mirrors reference Tcl's `Tcl_GetIntFromObj` behavior

**Upvar:**
- Top-level `upvar #0` / `upvar 0` (no proc frame) now creates global-scope `VAR_LINK` aliases in the root namespace instead of failing
- Variable traces now fire correctly when accessed through `upvar` aliases, reporting the alias name to the callback
- Trace cleanup through upvar aliases no longer causes stale unset callbacks

### Array Traces

Implements `array`-op variable traces: traces now fire when variables are accessed via the `array` command (get/set/names/size/exists/unset) but not on ordinary element reads/writes, matching reference Tcl behavior.

### Runtime Changes

- `ensure_stdlib_index_loaded()` populates `::auto_index` from `$TCL_LIBRARY/tclIndex` on first unknown command, enabling fallback auto-loading
- Frame-local variable trace firing through upvar aliases
- Global alias link support in namespace variable table

## Validation

- Added `tests/test_wasm_autoload.py` covering stdlib auto-loading and prelude bundling with synthetic and real library trees
- Added `tests/test_wasm_var_trace.py` tests for array traces and upvar trace firing
- Extended `tests/test_wasm_execution.py` with uplevel edge case tests (minus-zero, own-frame targeting, negative level errors)
- Updated baseline expectations: `uplevel` and `upvar` test suites now pass completely (57/57 and 62/70 respectively, with 8 skipped)
- All existing tests continue to pass

## Compiler/KCS Checklist

- [x] This change alters compiler behavior (adds stdlib bundling pass)
- [x] Updated KCS notes (new `docs/kcs/compiler/stdlib_prelude.md`)
- [x] Linked from `docs/kcs/compiler/README.md` and `docs/kcs/README.md`

https://claude.ai/code/session_01UyaMiKB9DZDLswmLY45QDe

### CODE REVIEW

File: runtime/zig/interp/tcl_ns.zig [hunk 1]
```diff
@@ -909,6 +909,34 @@ pub fn var_set_scalar(v_addr: u32, obj_handle: u32) void {
...
+/// ``var_resolve_link`` (tcltest upvar-7.1).  Re-aliasing an existing
+/// link simply re-points it.
+pub fn global_alias_link(local: i32, target: i32) void {
+    const ln = obj.obj_ensure_string(local);
+    const tn = obj.obj_ensure_string(target);
+    if (ln.len == 0 or ln.ptr == 0 or tn.len == 0 or tn.ptr == 0) return;
+    const lk = strip_global_prefix(ln.ptr, ln.len);
+    const tk = strip_global_prefix(tn.ptr, tn.len);
+    const target_var = ns_var_create(ns_root(), tk.ptr, tk.len);
+    if (target_var == 0) return;
```
[the remark this answers is not in the document]
Comment (bitwisecook) [reply runtime/zig/interp/tcl_ns.zig:949]: Fixed the silent divergence (4e3962a): `global_alias_link` now detects an array-element target (`arr(k)`) and skips creating the scalar `VAR_LINK`, so it no longer points at a phantom `arr(k)` scalar that diverges from the array directory `$arr(k)` uses.

Full correctness here (making `uv` actually alias element `k` of the global array) needs the same element-alias descriptor the in-proc path uses (`split_array_elem_name` + `resolve_global_array_elem_*`), but at the script top level there's no call frame to hold that descriptor, and array elements aren't stored as addressable `Var`s a `VAR_LINK` could target. So top-level `upvar` to an array element is left **unaliased** (the pre-feature behaviour) rather than wrongly linked — `arr(k)` keeps its value. The scalar case the feature targeted (`upvar #0 x uv`, upvar-7.1) is unaffected. Regression test added (`test_top_level_upvar_array_element_does_not_link_wrong_storage`). A proper global element-alias mechanism is a follow-up.

---
_Generated by [Claude Code](https://claude.ai/code)_
